services/QuillLMS/spec/services/serialize_activity_health_spec.rb

```diff
@@ -0,0 +1,92 @@
+
+  before do
+    ENV['DEFAULT_URL'] = 'https://quill.org'
+    ENV['CMS_URL'] = 'https://cms.quill.org'
+    stub_request(:get, "#{ENV['CMS_URL']}/questions/#{question.uid}/question_dashboard_data")
+      .to_return(status: 200, body: { percent_common_unmatched: 50,  percent_specified_algos: 75}.to_json, headers: {})
+    stub_request(:get, "#{ENV['CMS_URL']}/questions/#{another_question.uid}/question_dashboard_data")
+      .to_return(status: 200, body: { percent_common_unmatched: 100,  percent_specified_algos: 75}.to_json, headers: {})
+  end
+
+  it 'gets the correct basic data for that activity' do
+    data = SerializeActivityHealth.new(activity).data
+    expect(data[:name]).to eq(activity.name)
+    expect(data[:url]).to eq("https://quill.org/connect/#/admin/lessons/#{activity.uid}")
+    expect(data[:flag]).to eq(activity.flag.to_s)
+    expect(data[:activity_categories]).to eq(activity.activity_categories.pluck(:name))
+    expect(data[:content_partners]).to eq([content_partner.name])
+    expect(data[:tool]).to eq("connect")
+    expect(data[:activity_packs]).to eq(activity.unit_templates.pluck(:name))
+    expect(data[:diagnostics]).to eq([diagnostic.name])
+  end
+
+  it 'calculates the number plays in the last three months' do
+    UnitActivity.where(activity: activity).destroy_all
+    unit = create(:unit)
+    create(:classroom_unit, unit: unit, created_at: Date.today - 1.year)
+    create(:unit_activity, unit: unit, activity: activity)
+    data = SerializeActivityHealth.new(activity).data
+    expect(data[:recent_plays]).to eq(2)
+  end
+
+  it 'returns nil for recent_plays if the last assignment was less than 3 months ago' do
+    UnitActivity.where(activity: activity).destroy_all
+    unit = create(:unit)
+    create(:classroom_unit, unit: unit, created_at: Date.today - 1.month)
+    create(:unit_activity, unit: unit, activity: activity)
+    data = SerializeActivityHealth.new(activity).data
+    expect(data[:recent_plays]).to eq(nil)
+  end
+
+  it 'calculates averages and standard deviation using prompt data' do
+    data = SerializeActivityHealth.new(activity).data
+    expect(data[:avg_common_unmatched]).to eq(75)
+    expect(data[:avg_difficulty]).to eq(1.84)
+    expect(data[:standard_dev_difficulty]).to eq(0.84)
+  end
+
+  it 'calculates the average length of time to finish the activity' do
+    data = SerializeActivityHealth.new(activity).data
+    expect(data[:avg_mins_to_complete]).to eq(11.67)
+  end
+
+  it 'calculates the data without nil erroring for a brand new activity with no assignments or attributes' do
+    new_activity = create(:activity, activity_classification_id: connect.id)
+    new_activity.update(activity_categories: [], content_partners: [], units: [])
+    new_activity.save
+
+    data = SerializeActivityHealth.new(new_activity).data
+    expect(data[:activity_categories]).to eq([])
```

Comment:
For future reference: you can construct expected result objects and use those to compare, removing the need for so many expectations. You can even have the expectation objects be a strict subset of the actual output, for maximum flexibility. [example](https://github.com/empirical-org/Empirical-Core/blob/develop/services/QuillLMS/spec/lib/rule_feedback_history_spec.rb#L47)